### PR TITLE
Update with latest changes from IBM/simpool

### PR DESCRIPTION
**Pull request by davidbeckingsale**

### CODE REVIEW

File: AllocatorTest.hpp [hunk 1]
```diff
@@ -81,7 +81,7 @@ struct STLAllocator {
   typedef T value_type;
   typedef std::size_t size_type;
 
-  typedef DynamicPoolAllocator<AllocatorType> PoolType;
+  typedef DynamicSizePool<AllocatorType> PoolType;
```
Comment (mcfadden8) [inline AllocatorTest.hpp:84]: We will need to make sure and change all of our using code to use this new name.

File: DynamicSizePool.hpp [hunk 2]
```diff
@@ -24,8 +27,8 @@ class DynamicPoolAllocator
   };
 
   // Allocator for the underlying data
-  typedef FixedPoolAllocator<struct Block, IA, (1<<6)> BlockAlloc;
-  BlockAlloc blockAllocator;
+  typedef FixedSizePool<struct Block, IA, IA, (1<<6)> BlockPool;
```
Comment (mcfadden8) [inline DynamicSizePool.hpp:28]: I'm not sure I understand this. Shouldn't this be "MA, IA" instead of "IA, IA"?

Comment (davidbeckingsale) [reply DynamicSizePool.hpp:28]: The new FixedPool can use different allocators for the meta-data and the actual data (e.g. for storing data in GPU memory). In this case, we want to use the same allocator for both. So I think this is correct.

Comment (mcfadden8) [reply DynamicSizePool.hpp:28]: I'm not sure I see why FixedPool has use cases for separate allocators of meta-data and actual data and DynamicSizePool does not?  What is the unique use case(s) that FixedPool has that DynamicSizePool does not with respect to this?

Comment (davidbeckingsale) [reply DynamicSizePool.hpp:28]: FixedPool was storing the metadata and actual data all in the same allocation. DynamicPool does not (they are already separate).

Comment (mcfadden8) [reply DynamicSizePool.hpp:28]: Okay.  I see.

**Comment by jdahm** [comment]
Let me know if I can help with this by changing anything in simpool itself that doesn't look right. I know I need to add better testing still, and I do want to incorporate @mcfadden8 point about block splitting. That's all on the todo list.

**Comment by davidbeckingsale** [comment]
@mcfadden8 - this should be good to go. Note that I have added a new pool that is basically a copy of the fixed-size pool, modified to use an AllocationStrategy. This allows us to still use the old one without an AllocationStrategy (faster) inside the dynamic pool.

**Comment by davidbeckingsale** [comment]
@mcfadden8 - this is ready and tested.

File: Makefile [hunk 3]
```diff
@@ -1,8 +1,8 @@
--include config.h
+# config.mk can overwrite these defaults
+-include config.mk
 
-# config.h can overwrite these defaults
 CXX ?= clang++
-CXXFLAGS ?= -O3
+CXXFLAGS ?= -O1 -g
```
Comment (mcfadden8) [inline Makefile:5]: Why did this change from -O3 to -O1?

**Comment by jdahm** [comment]
Probably because I was testing something and forgot to set it back. I'll make that change now. EDIT: pushed.

File: Makefile (hunk 3 above)
Comment (davidbeckingsale) [reply Makefile:5]: No idea, but it doesn't impact us.

Comment (mcfadden8) [reply Makefile:5]: Good point @davidbeckingsale, I should have noticed that this was in a Makefile.

**Review by mcfadden8 — APPROVED** [review]
Looks good

**Comment by mcfadden8** [comment]
@davidbeckingsale, can this branch be removed now that it has been merged?

**Comment by davidbeckingsale** [comment]
Sure!

**Pull request merged**